web/vtadmin/src/components/routes/shard/Advanced.tsx

```diff
@@ -0,0 +1,413 @@
+                    `Successfully failed over shard ${params.shard} to tablet ${formatAlias(
+                        plannedReparentTablet?.tablet?.alias
+                    )}.`
+                );
+                setIsOpen(true);
+                setEvents(result.events);
+            },
+            onError: (error) =>
+                warn(`There was an error failing over shard ${params.shard} to ${plannedReparentTablet}: ${error}`),
+        }
+    );
+
+    // emergencyReparent parameters
+    const [emergencyReparentTablet, setEmergencyReparentTablet] = useState<vtadmin.Tablet | null>(null);
+    const emergencyReparentMutation = useEmergencyFailoverShard(
+        {
+            clusterID: params.clusterID,
+            keyspace: params.keyspace,
+            shard: params.shard,
+            new_primary: emergencyReparentTablet as vtadmin.Tablet,
+        },
+        {
+            onSuccess: (result) => {
+                setDialogTitle(`Emergency Failover`);
+                setDialogDescription(
+                    `Successfully failed over ${params.shard} to tablet ${formatAlias(
+                        emergencyReparentTablet?.tablet?.alias
+                    )}.`
+                );
+                setIsOpen(true);
+                setEvents(result.events);
+            },
+            onError: (error) =>
+                warn(
+                    `There was an error failing over shard ${params.shard} to ${formatAlias(
+                        emergencyReparentTablet?.tablet?.alias
+                    )}: ${error}`
+                ),
+        }
+    );
+
+    if (kq.error) {
+        return (
+            <div className="items-center flex flex-column text-lg justify-center m-3 text-center max-w-[720px]">
+                <span className="text-xl">😰</span>
+                <h1>An error occurred</h1>
+                <code>{(kq.error as any).response?.error?.message || kq.error?.message}</code>
+                <p>
+                    <Link to="/keyspaces">← All keyspaces</Link>
+                </p>
+            </div>
+        );
+    }
+
+    return (
+        <>
+            <Dialog
+                className="min-w-[500px]"
+                isOpen={isOpen}
+                onClose={() => setIsOpen(false)}
+                title={dialogTitle}
+                hideCancel
+                confirmText="Dismiss"
+            >
+                <>
+                    <div className="mt-8 mb-4">{dialogDescription}</div>
+                    <div className="mb-2 font-bold">Log</div>
+                    <div className="bg-gray-100 p-4 overflow-scroll max-h-[200px]">
+                        {events.map((e) => (
+                            <div className="font-mono text-sm whitespace-nowrap">
+                                [{new Date((e.time?.seconds as number) * 1000).toISOString()} {e.file}:{e.line}]{' '}
+                                {e.value}
+                            </div>
```

Comment:
This could be extracted out into an `EventLogEntry` component or something to slim down this component a little bit.